util/codec/codec.go

```diff
@@ -352,6 +352,240 @@ func encodeHashChunkRowIdx(sc *stmtctx.StatementContext, row chunk.Row, tp *type
 	return
 }
 
+func writeColumn(w io.Writer, flag, data []byte) (err error) {
+	_, err = w.Write(flag)
+	if err != nil {
+		return
+	}
+	_, err = w.Write(data)
+	return
+}
+
+// HashChunkColumns writes the encoded value of each row's column, which of index `colIdx`, to w.
+// `nullBitmap` used to save the rows whose column of index `colIdx` is `Null`.
+func HashChunkColumns(sc *stmtctx.StatementContext, w []io.Writer, chk *chunk.Chunk, tp *types.FieldType, colIdx int, buf, nullBitmap []byte) (err error) {
+	var (
+		b    []byte
+		rows int
+		row  chunk.Row
+	)
+	rows = chk.NumRows()
+	switch tp.Tp {
+	case mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24, mysql.TypeLong, mysql.TypeLonglong, mysql.TypeYear:
+		for i := 0; i < rows; i++ {
+			row = chk.GetRow(i)
+			if row.IsNull(colIdx) {
+				buf[0], b = NilFlag, nil
+				nullBitmap[i/8] |= 1 << uint(i%8)
+			} else {
+				buf[0] = varintFlag
+				if mysql.HasUnsignedFlag(tp.Flag) {
+					if integer := row.GetInt64(colIdx); integer < 0 {
+						buf[0] = uvarintFlag
+					}
+				}
+				b = row.GetRaw(colIdx)
+			}
+
+			err = writeColumn(w[i], buf, b)
```

Comment:
We can put `writeColumn` out of the `switch` statement to make it simple.

Reply:
I can not figure out how to do this directly.